src/runmode-unix-socket.c

```diff
@@ -310,41 +390,73 @@ static TmEcode UnixSocketPcapFilesCheck(void *data)
 
     PcapFiles *cfile = TAILQ_FIRST(&this->files);
     TAILQ_REMOVE(&this->files, cfile, next);
-    SCLogInfo("Starting run for '%s'", cfile->filename);
-    unix_manager_file_task_running = 1;
+
+    unix_manager_pcap_task_running = 1;
     this->running = 1;
-    if (ConfSet("pcap-file.file", cfile->filename) != 1) {
-        SCLogError(SC_ERR_INVALID_ARGUMENTS,
-                "Can not set working file to '%s'", cfile->filename);
+
+    if (ConfSetFinal("pcap-file.file", cfile->filename) != 1) {
+        SCLogError(SC_ERR_INVALID_ARGUMENT, "Can not set working file to '%s'",
+                   cfile->filename);
         PcapFilesFree(cfile);
         return TM_ECODE_FAILED;
     }
-    if (cfile->output_dir) {
-        if (ConfSet("default-log-dir", cfile->output_dir) != 1) {
-            SCLogError(SC_ERR_INVALID_ARGUMENTS,
-                    "Can not set output dir to '%s'", cfile->output_dir);
+
+    int set_res = 0;
+    if(cfile->continuous) {
+        set_res = ConfSetFinal("pcap-file.continuous", "true");
+    } else {
+        set_res = ConfSetFinal("pcap-file.continuous", "false");
+    }
+    if(set_res != 1) {
+        SCLogError(SC_ERR_INVALID_ARGUMENT, "Can not set continuous");
+        PcapFilesFree(cfile);
+        return TM_ECODE_FAILED;
+    }
+
+    if(cfile->delay > 0) {
+        char tstr[32];
+        snprintf(tstr, sizeof(tstr), "%ld", cfile->delay);
+        if (ConfSetFinal("pcap-file.delay", tstr) != 1) {
```

Comment:
Probably not. Never thought of the use case of setting values after initialization. Probably best avoided.

Reply:
I think the existing code does some of this as well. It's also one of the reasons I did the 'prefix' loading for tenants. Then at least it's adding stuff in a new namespace.

I don't think there is a lot of code that does Conf calls after startup, so perhaps we can just wrap a lock around things.

Another problem is the potential for consumers of the api to keep reference to data in the api. If that does away we'll end up with dangling pointers. Ideally the api should allow for it.

Reply:
Edit: "Ideally the api should NOT allow for it"

Reply:
Yeah the existing code is doing the same conf settings https://github.com/OISF/suricata/blob/master/src/runmode-unix-socket.c#L316.

It's sort of kind of okay since the threads are then getting kicked off to read the values. 

Is it ok for this PR to keep this in place, or is there an alternative?

Reply:
I guess we can let it go in, as it should be reasonably safe. But if you're up for it, I would prefer a thread safe solution. Something like a mutex protected TAILQ that we add jobs to so we wouldn't need to touch the Conf API at all.

Reply:
Due to the size of the PR, created https://redmine.openinfosecfoundation.org/issues/2296 to look into a way to not touch the Conf API.